lib/bambora/bank/batch_report_resource.rb

```diff
@@ -61,9 +61,14 @@ def show(report_data)
       private
 
       def add_messages_to_response(response)
+        # bambora can return null or empty record results
+        records = response.dig(:response, :record)
+        return response if records.nil?
```

Comment:
Minor naming thing here that is a bit confusing in the response from Bambora. There are two nodes `records` (plural) that has a count of the total number of records and `record` (singular) that contains all the records.

It would be more clear to either do:

```suggestion
        record = response.dig(:response, :record)
        return response if record.nil?
```

or

```suggestion
        total = response.dig(:response, :records, :total)
        return response if total.zero?
```

This way the variable name matches the node name.

Reply:
I was definitely scratching my head over Bambora's API about this. I think a minor change like this could help with that understanding.

Reply:
- [x] Updated this to remove the unneeded variable

Thanks for the feedback. Marking as resolved.